src/MOL_discretization.jl

```diff
@@ -6,29 +6,113 @@ struct MOLFiniteDifference{T} <: DiffEqBase.AbstractDiscretization
     MOLFiniteDifference(args...;order=2) = new{typeof(args[1])}(args[1],order)
 end
 
+function throw_bc_err(bc)
+    throw(BoundaryConditionError(
+        "Could not recognize boundary condition '$(bc.lhs) ~ $(bc.rhs)'"
+    ))
+end
 # Get boundary conditions from an array
+# The returned boundary conditions will be the three coefficients (α, β, γ)
+# for a RobinBC of the form α*u(0) + β*u'(0) = γ
+# These are created automatically from the form of the boundary condition that is
+# passed, i.e.
+# u(t, 0) ~ γ                           ---> (1, 0, γ)
+# Dx(u(t, 0)) ~ γ                       ---> (0, 1, γ)
+# α * u(t, 0)) + β * Dx(u(t, 0)) ~ γ    ---> (α, β, γ)
+# In general, all of α, β, and γ could be Expr 
 function get_bcs(bcs,tdomain,domain)
     lhs_deriv_depvars_bcs = Dict()
     num_bcs = size(bcs,1)
     for i = 1:num_bcs
-        var = operation(bcs[i].lhs)
-        if var isa Sym
-            var = var.name
-            if !haskey(lhs_deriv_depvars_bcs,var)
-                lhs_deriv_depvars_bcs[var] = Array{Expr}(undef,3)
-            end
-            j = 0
-            if isequal(bcs[i].lhs.args[1],tdomain.lower) # u(t=0,x)
-                j = 1
-            elseif isequal(bcs[i].lhs.args[2],domain.lower) # u(t,x=x_init)
-                j = 2
-            elseif isequal(bcs[i].lhs.args[2],domain.upper) # u(t,x=x_final)
-                j = 3
+        lhs = bcs[i].lhs
+        # Extract the variable from the lhs
+        if operation(lhs) isa Sym
+            # Dirichlet boundary condition
+            var = operation(lhs).name
+            α = 1.0
+            β = 0.0
+            bc_args = lhs.args
+        elseif operation(lhs) isa Differential
+            # Neumann boundary condition
+            var = operation(lhs.args[1]).name

```

Comment:
Order of the derivative needs to be checked here?

Reply:
ok, what's the command to get the order?

Reply:
I think you want to check if `operation(operation(lhs)) isa Differential` and throw an error directly. That should be good for now since it has to be first order for Robin.